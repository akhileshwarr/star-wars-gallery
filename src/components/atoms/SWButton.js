import Button from '@material-ui/core/Button';
import React from 'react';
import PropTypes from 'prop-types';

export const SWButton = ({
    children,
    style = {},
    handleSWClick = () => {},
    disabled = false
 }) => {

    return (
        <Button
        size="large"
        variant="contained"
        color="primary"
        style = {{...style}}
        onClick = {handleSWClick}
        disabled = {disabled}
      >
          {children}
          </Button>
    );
  };

  SWButton.propTypes = {
    children: PropTypes.element,
    style: PropTypes.object,
    handleSWClick:PropTypes.func,
    disabled: PropTypes.bool
  };
  
  SWButton.defaultProps = {
    children: null,
    style: {},
    handleSWClick:null,
    disabled: false

  };
  