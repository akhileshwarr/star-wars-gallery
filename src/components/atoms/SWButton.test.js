import React from 'react';
import renderer from 'react-test-renderer';
import { SWButton } from './SWButton.js';

test('/SWButton renders snapshot', () => {
    const tree = renderer.create(<SWButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });