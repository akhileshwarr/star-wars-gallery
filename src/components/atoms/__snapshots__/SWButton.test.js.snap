// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Container renders snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
  disabled={false}
  onBlur={[Function]}
  onClick={null}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={Object {}}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  />
</button>
`;

exports[`/SWButton renders snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
  disabled={false}
  onBlur={[Function]}
  onClick={null}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={Object {}}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  />
</button>
`;
